File: app/src/main/java/net/imatruck/dsmanager/tasks/DSTaskResumeTask.java
package net.imatruck.dsmanager.tasks;

import android.app.Activity;
import android.os.AsyncTask;
import android.widget.TextView;

import net.imatruck.dsmanager.R;
import net.imatruck.dsmanager.models.DSTaskResumeBase;
import net.imatruck.dsmanager.models.DSTaskResumeData;
import net.imatruck.dsmanager.utils.SynologyDSTaskError;

import java.io.IOException;
import java.util.List;
import java.util.Locale;

import retrofit2.Call;


public class DSTaskResumeTask extends AsyncTask<Call<DSTaskResumeBase>, Void, DSTaskResumeBase> {

    private Activity context;

    public DSTaskResumeTask(Activity context) {
        this.context = context;
    }

    @SafeVarargs
    @Override
    protected final DSTaskResumeBase doInBackground(Call<DSTaskResumeBase>... calls) {

        DSTaskResumeBase dsTaskResumeBase;
        try {
            dsTaskResumeBase = calls[0].execute().body();
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }

        return dsTaskResumeBase;
    }

    @Override
    protected void onPostExecute(DSTaskResumeBase dsInfo) {
        if (dsInfo != null) {
            boolean success = dsInfo.isSuccess();
            TextView debugTextView = (TextView) context.findViewById(R.id.debug_api_text);

            if (success) {
                List<DSTaskResumeData> data = dsInfo.getData();
                String text = "";

                for (DSTaskResumeData dsTaskResumeData : data) {
                    text += String.format(Locale.getDefault(),
                            "%s: %s\n",
                            dsTaskResumeData.getId(),
                            context.getString(
                                    SynologyDSTaskError.getMessageId(dsTaskResumeData.getError()))
                    );
                }
                debugTextView.setText(text);
            } else {
                String text = context.getString(
                        SynologyDSTaskError.getMessageId(dsInfo.getError().getCode()));
                debugTextView.setText(text);
            }
        }
    }
}

File: app/src/main/java/net/imatruck/dsmanager/tasks/DSTaskResumeTask.kt
package net.imatruck.dsmanager.tasks

import android.app.Activity
import android.os.AsyncTask
import android.widget.TextView
import net.imatruck.dsmanager.R
import net.imatruck.dsmanager.models.DSTaskResumeBase
import net.imatruck.dsmanager.utils.SynologyDSTaskError
import retrofit2.Call
import java.io.IOException
import java.util.*


class DSTaskResumeTask(private val context: Activity) : AsyncTask<Call<DSTaskResumeBase>, Void, DSTaskResumeBase>() {

    @SafeVarargs
    override fun doInBackground(vararg calls: Call<DSTaskResumeBase>): DSTaskResumeBase? {

        val dsTaskResumeBase: DSTaskResumeBase
        try {
            dsTaskResumeBase = calls[0].execute().body()
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }

        return dsTaskResumeBase
    }

    override fun onPostExecute(dsInfo: DSTaskResumeBase?) {
        if (dsInfo != null) {
            val success = dsInfo.isSuccess
            val debugTextView = context.findViewById<TextView>(R.id.debug_api_text)

            if (success) {
                val data = dsInfo.data
                var text = ""

                for (dsTaskResumeData in data) {
                    text += String.format(Locale.getDefault(),
                            "%s: %s\n",
                            dsTaskResumeData.id,
                            context.getString(
                                    SynologyDSTaskError.getMessageId(dsTaskResumeData.error))
                    )
                }
                debugTextView.text = text
            } else {
                val text = context.getString(
                        SynologyDSTaskError.getMessageId(dsInfo.error.code))
                debugTextView.text = text
            }
        }
    }
}
